src/shared/edit-util.c

```diff
@@ -18,19 +18,93 @@
 #include "strv.h"
 #include "tmpfile-util.h"
 
-void edit_file_free_all(EditFile **f) {
-        if (!f || !*f)
-                return;
+void edit_file_context_done(EditFileContext *context) {
+        int r;
+
+        assert(context);
+
+        FOREACH_ARRAY(i, context->files, context->n_files) {
+                if (i->temp) {
+                        (void) unlink(i->temp);
+                        free(i->temp);
+                }
```

Comment:
i->temp = unlink_and_free(i->temp)

Reply:
This has been applied in https://github.com/systemd/systemd/pull/26776 :)